network/dag/notifier.go

```diff
@@ -330,21 +347,24 @@ func (p *notifier) Notify(event Event) {
 func (p *notifier) retry(event Event) {
 	delay := p.retryDelay
 	initialCount := event.Retries + 1
+	attempts := maxRetries - uint(initialCount)
+	if attempts >= maxRetries {
+		attempts = 0
+	}
 
 	for i := 0; i < initialCount; i++ {
 		delay *= 2
 	}
 
 	go func(ctx context.Context) {
-		// also an initial delay
-		time.Sleep(delay)
 		err := retry.Do(func() error {
 			return p.notifyNow(event)
 		},
-			retry.Attempts(maxRetries-uint(initialCount)),
+			retry.Attempts(attempts),
```

Comment:
`attempts==0` means infinite Attempts

Reply:
anything <0 and >max will just return the retry.